### PR TITLE
Add scheduled sampling howto for seq2seq example.

### PR DESCRIPTION
**Pull request by adarob**

### CODE REVIEW

**Comment by jheek** [comment]
you probable want to add your howto to the docs in `docs/howtos.rst`.

**Comment by adarob** [comment]
@avital @marcvanzee

File: howtos/diffs/scheduled-sampling.diff [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+-def train_step(optimizer, batch, rng):
++def train_step(optimizer, batch, sample_probability, rng):
+   """Train one step."""
+   labels = batch['answer'][:, 1:]  # remove '=' start token
+ 
+   def loss_fn(model):
+     """Compute cross-entropy loss."""
+-    with nn.stochastic(rng):
+-      logits, _ = model(batch['query'], batch['answer'])
++    logits, _ = model(
```
Comment (avital) [inline howtos/diffs/scheduled-sampling.diff:85]: Is `with nn.stochastic(rng)` no longer necessary? That would surprise me...

Comment (avital) [reply howtos/diffs/scheduled-sampling.diff:85]: (Or if so, it should be removed from the base example. I suppose we could do that in a separate commit)

Comment (adarob) [reply howtos/diffs/scheduled-sampling.diff]: Fixed and tested.

**Comment by avital** [comment]
Super cool!

**Pull request merged**